src/main/java/com/codeless/api/automation/storage/RateLimitStorage.java

```diff
@@ -1,11 +1,16 @@
 package com.codeless.api.automation.storage;
 
+import java.util.LinkedHashMap;
 import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
 
 public final class RateLimitStorage {
 
-  private static final Map<String, RateLimitData> STORAGE = new ConcurrentHashMap<>();
+  private static final LinkedHashMap<String, RateLimitData> STORAGE = new LinkedHashMap<>(){
```

Comment:
I think it is bad practice to specify concrete implementation on the left side of declaration.